src/anchor-navigation/scroll-spy.tsx

```diff
@@ -0,0 +1,79 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import { useCallback, useEffect, useState } from 'react';
+
+const isBrowser = typeof window !== 'undefined';
+
+export default function useScrollSpy({
```

Comment:
Looks like this hook assumes that the hrefs, and by extension the anchors passed in in the props, are sorted by order of appearance in the page. Is this a safe assumption? Or should be either a) make it explicit in the API description or b) let the component sort them

Reply:
Good point. I'd be for `a)`.
For `b)`, I can imagine this could add some computational overhead, especially if the list of `hrefs` is long or if the DOM is complex. Also, I feel it becomes _too much_ magic.